Bluetooth: Mesh: Trim _USES_MBEDTLS_PSA selects

HaavardRei opened the pull request:
Removes legacy configuration options from BT_MESH_USES_MBEDTLS_PSA. The same coverage is provided by existing PSA_WANT_... selects.

alxelax commented on subsys/bluetooth/mesh/Kconfig:
```diff
@@ -1477,25 +1477,21 @@ config BT_MESH_USES_MBEDTLS_PSA
 	select PSA_WANT_ALG_HMAC
 	select PSA_WANT_ALG_SHA_256
 	select PSA_WANT_ALG_ECDH
 	select PSA_WANT_ALG_ECDSA
```
I guess `PSA_WANT_ALG_ECDSA` is not required either. This is for signatures those Mesh doesn't use. Previously, mbedtls united it with ECDH.

HaavardRei replied:
Removed it now

HaavardRei committed:
Bluetooth: Mesh: Trim _USES_MBEDTLS_PSA selects

Removes legacy configuration options. The same coverage is provided by
existing PSA_WANT_... selects.

Signed-off-by: Håvard Reierstad <haavard.reierstad@nordicsemi.no>

jhedberg commented:
Looks like there's some unrelated CI breakage: 
```
INFO: Error found: storage.stream_flash on nrf52840dk/nrf52840 (Not in testsuite platform allow list but is one of the integration platforms)
```

I've seen the same failure in other PRs too.

jhedberg commented:
> Looks like there's some unrelated CI breakage:
> 
> ```
> INFO: Error found: storage.stream_flash on nrf52840dk/nrf52840 (Not in testsuite platform allow list but is one of the integration platforms)
> ```
> 
> I've seen the same failure in other PRs too.

Hotfix available here: #82222
